File: src/brs/http/GetTime.java
package brs.http;

import brs.services.TimeService;
import org.json.simple.JSONObject;
import org.json.simple.JSONStreamAware;

import javax.servlet.http.HttpServletRequest;

import static brs.http.common.ResultFields.TIME_RESPONSE;

public final class GetTime extends APIServlet.APIRequestHandler {

  private final TimeService timeService;

  GetTime(TimeService timeService) {
    super(new APITag[]{APITag.INFO});
    this.timeService = timeService;
  }

  @Override
  JSONStreamAware processRequest(HttpServletRequest req) {
    JSONObject response = new JSONObject();
    response.put(TIME_RESPONSE, timeService.getEpochTime());

    return response;
  }

}

File: src/main/java/brs/http/GetTime.kt
package brs.http

import brs.services.TimeService
import com.google.gson.JsonElement
import com.google.gson.JsonObject

import javax.servlet.http.HttpServletRequest

import brs.http.common.ResultFields.TIME_RESPONSE

internal class GetTime internal constructor(private val timeService: TimeService) : APIServlet.JsonRequestHandler(arrayOf(APITag.INFO)) {

    internal override fun processRequest(req: HttpServletRequest): JsonElement {
        val response = JsonObject()
        response.addProperty(TIME_RESPONSE, timeService.epochTime)

        return response
    }

}
